cmd/reconciler-manager/main.go

```diff
@@ -77,18 +76,22 @@ func main() {
 		setupLog.Error(err, "unable to start manager")
 		os.Exit(1)
 	}
+	dynamicClient, err := dynamic.NewForConfig(mgr.GetConfig())
```

Comment:
I'm surprised we don't already have a dynamic client in the reconciler-manager...

I'm also wondering if we should be injecting the rest.Config, instead of letting the controller manager build it. I think this means we aren't using the same timeout and QPS settings used by the reconciler...

We might want to build the config with restconfig.NewRestConfig and pass it into the ctrl.Options, and then also use it to build the dynamic client (assuming the controller manager doesn't already build one we can use instead).